src/plonk/keygen.rs

```diff
@@ -267,12 +270,34 @@ where
     let l0 = vk.domain.lagrange_to_coeff(l0);
     let l0 = vk.domain.coeff_to_extended(l0, Rotation::cur());
 
+    // Compute l_cover(X) which evaluates to 1 for each blinding factor row
+    // and 0 otherwise over the domain.
+    let mut l_cover = vk.domain.empty_lagrange();
+    for evaluation in l_cover[..].iter_mut().rev().take(cs.blinding_factors()) {
+        *evaluation = C::Scalar::one();
+    }
+    let l_cover = vk.domain.lagrange_to_coeff(l_cover);
+    let l_cover = vk.domain.coeff_to_extended(l_cover, Rotation::cur());
+
+    // Compute l_last(X) which evaluates to 1 on the first inactive row (just
+    // before the blinding factors) and 0 otherwise over the domain
+    let mut l_last = vk.domain.empty_lagrange();
+    *(l_last[..]
+        .iter_mut()
+        .rev()
+        .nth(cs.blinding_factors())
+        .unwrap()) = C::Scalar::one();
```

Comment:
Can this be expressed more simply as something like `l_last[params.n - cs.blinding_factors() - 1] = C::Scalar::one();`?